site/components/form/DateSelector.tsx

```diff
@@ -76,17 +78,36 @@ const DateSelector = <T extends object>({
     optional = false,
     disablePast,
     stateUpdater,
+    reset = false,
+    showError = false,
 }: DateSelectorProps<T>): ReactElement => {
     const [dateValue, setDateValue] = useState<Date | null>(!!disabled || !value ? null : new Date(value));
     const [errors, setErrors] = useState<ErrorInfo[]>(initialErrors);
 
+    useEffect(() => {
+        if (reset) {
+            setDateValue(null);
+        }
+    }, [reset]);
+
     useEffect(() => {
         if (disabled) {
             setErrors([]);
             setDateValue(null);
         }
     }, [disabled]);
 
+    useEffect(() => {
```

Comment:
Selecting the date using date picker after an error does not remove the error message